.github/workflows/check-catalyst.yaml

```diff
@@ -234,6 +242,15 @@ jobs:
         enableCrossOsArchive: True
         fail-on-cache-miss: True
 
+    - name: Get Cached MHLO Source
+      id: cache-mhlo-source
+      uses: actions/cache@v3
+      with:
+        path: mlir/mlir-hlo
+        key: ${{ runner.os }}-mhlo-${{ needs.constants.outputs.mhlo_version }}-default-source
+        enableCrossOsArchive: True
```

Comment:
Why would you remove it? The source cache can definitely be shared across platforms, no?

Reply:
We're not using multiple platforms at the moment. 

Where is the MHLO source used by the way?

Reply:
No harm on keeping this as is. It just caught my eyes and wanted to ensure that there's some reasons behind this flag and everyone is aware of :smile:

Reply:
Yes I put it there explicitly because it seemed like the correct declaration for this cache :)

We may or may not use it in the future, but if we do, it's nice that it already has the correct properties, which could be easy to miss when adding a windows action for example.